cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java

```diff
@@ -225,6 +253,36 @@ public Void apply(AppMds mds) throws Exception {
       }
     });
 
+    if (id.getType() == ProgramType.WORKFLOW && runStatus == ProgramRunStatus.COMPLETED) {
+      final RunRecordMeta run = getRun(id, pid);
+      Id.Application app = id.getApplication();
+      ApplicationSpecification appSpec = getApplication(app);
+      WorkflowSpecification workflowSpec = appSpec.getWorkflows().get(id.getId());
+      final Map<String, WorkflowNode> nodeIdMap = workflowSpec.getNodeIdMap();
+
+      final List<WorkflowDataset.ActionRuns> actionRunsList = new ArrayList<>();
+
+      for (Map.Entry<String, String> entry : run.getProperties().entrySet()) {
+        if (!entry.getKey().equals("workflowToken")) {
```

Comment:
So the reason behind doing this is that when the key is "workflowToken", the value is a monstrous object that I have no use of. So I could use an else but my if would practically be a continue;